build.sbt

```diff
@@ -1,26 +1,34 @@
 
 lazy val commonSettings = Seq(
   scalaVersion := "2.11.8",
-  addCompilerPlugin("org.spire-math" %% "kind-projector" % "0.8.0")
+  addCompilerPlugin("org.spire-math" %% "kind-projector" % "0.8.0"),
+  libraryDependencies += "org.scala-lang" %  "scala-reflect" % scalaVersion.value
 )
 
 lazy val core = project
   .in(file("modules/core"))
   .settings(commonSettings)
   .settings(
     libraryDependencies ++= Seq(
-      "org.tpolecat" %% "doobie-core"               % "0.3.0-M1", // uh, lame
-      "org.tpolecat" %% "doobie-contrib-postgresql" % "0.3.0-M1"
+      "org.scalaz"  %% "scalaz-core" % "7.2.4",
+      "com.chuusai" %% "shapeless"   % "2.3.1"
```

Comment:
The learning curve for shapeless seems steep, at least the introductions I've read are very confusing. If we need to use it, I think we should add training time for the rest of us.